platforms/android/library-compose/src/main/java/io/element/android/wysiwyg/compose/EditorStyledText.kt

```diff
@@ -18,27 +24,47 @@ import io.element.android.wysiwyg.display.MentionDisplayHandler
  * If it's spanned it will be rendered as is, otherwise it will go first to HtmlConverter.
  * Your might want to use HtmlConverter before the rendering to avoid the conversion at each recomposition.
  * @param modifier The modifier for the layout.
+ * @param resolveMentionDisplay A function to resolve the [TextDisplay] of a mention.
+ * @param resolveRoomMentionDisplay A function to resolve the [TextDisplay] of an `@room` mention.
  * @param style The styles to use for any customisable elements.
  */
 @Composable
 fun EditorStyledText(
     text: CharSequence,
     modifier: Modifier = Modifier,
-    mentionDisplayHandler: MentionDisplayHandler? = null,
+    resolveMentionDisplay: ((text: String, url: String) -> TextDisplay)? = null,
+    resolveRoomMentionDisplay: (() -> TextDisplay)? = null,
     style: RichTextEditorStyle = RichTextEditorDefaults.style(),
 ) {
-    val typeface = style.text.rememberTypeface()
+    val typeface by style.text.rememberTypeface()
+    var previousStyle by remember { mutableStateOf<RichTextEditorStyle?>(null) }
+    val mentionDisplayHandler = remember(resolveMentionDisplay, resolveRoomMentionDisplay) {
+        object : MentionDisplayHandler {
+            override fun resolveMentionDisplay(text: String, url: String): TextDisplay {
+                return resolveMentionDisplay?.invoke(text, url) ?: TextDisplay.Plain
+            }
+
+            override fun resolveAtRoomMentionDisplay(): TextDisplay {
+                return resolveRoomMentionDisplay?.invoke() ?: TextDisplay.Plain
+            }
+        }
+    }
     AndroidView(modifier = modifier, factory = { context ->
         EditorStyledTextView(context)
-    }, update = { view ->
-        view.setStyleConfig(style.toStyleConfig(view.context))
-        view.applyStyle(style)
-        view.typeface = typeface
-        view.mentionDisplayHandler = mentionDisplayHandler
-        if (text is Spanned) {
-            view.text = text
-        } else {
-            view.setHtml(text.toString())
+    }, update = remember(style != previousStyle, typeface, mentionDisplayHandler, text) {
+        { view ->
+            if (style != previousStyle) {
+                view.styleConfig = style.toStyleConfig(view.context)
+                view.applyStyleInCompose(style)
+                previousStyle = style
+            }
```

Comment:
I'm not sure about the implications of setting a variable from within the `update` block. Is it strictly necessary to prevent the style getting updated more than necessary given the `text` is unlikely to be getting updated frequently like it is with the editor itself?

Reply:
I'm removing this and trying to use a better API.